### ISSUE TITLE
Error Authenticating

### ISSUE THREAD

**Issue by Vannazux** [coinigy/WebsocketAPIClient#5]
After the last 2 days I am getting the error below to authenticate using Demo client.

Anything I should do?
```

2018-05-28T14:01:12.0943560-03:00 PureSocketClusterSocket..ctor: Creating new instance.
2018-05-28T14:01:12.1015984-03:00 PureWebSocket..ctor: Creating new instance.
2018-05-28T14:01:12.1576221-03:00 PureWebSocket.StartMonitor: Starting monitor.
2018-05-28T14:01:12.1657466-03:00 PureSocketClusterSocket.SetupEvents: Attaching events.
2018-05-28T14:01:12.1740893-03:00 PureWebSocket.StartMonitor: Entering monitor loop.
2018-05-28T14:01:12.1858208-03:00 PureSocketClusterSocket.Connect: Connect invoked.
2018-05-28T14:01:12.1993974-03:00 PureWebSocket.Connect: Connect called.
2018-05-28T14:01:12.3757440-03:00 PureWebSocket.StartMonitor: State changed from None to Connecting.
2018-05-28T14:01:12.3767976-03:00 PureSocketClusterSocket.Socket_OnStateChanged: State changed fomr None to Connecting.
Socket state changed from None to Connecting
2018-05-28T14:01:13.0662372-03:00 PureWebSocket.Connect: Starting tasks.
2018-05-28T14:01:13.0675525-03:00 PureWebSocket.StartListener: Starting listener.
2018-05-28T14:01:13.0684080-03:00 PureWebSocket.StartSender: Starting sender.
Connected = True
2018-05-28T14:01:13.0699858-03:00 PureWebSocket.Connect: Connect result: True, State Open
2018-05-28T14:01:13.1150330-03:00 PureWebSocket.StartSender: Entering sender loop.
2018-05-28T14:01:13.1247135-03:00 PureWebSocket.StartListener: Entering listener loop.
2018-05-28T14:01:13.1826584-03:00 PureWebSocket.StartMonitor: State changed from Connecting to Open.
2018-05-28T14:01:13.1826911-03:00 PureSocketClusterSocket.Socket_OnStateChanged: State changed fomr Connecting to Open.
Socket state changed from Connecting to Open
2018-05-28T14:01:13.1903572-03:00 PureSocketClusterSocket.Socket_OnOpened: OnOpened invoked.
2018-05-28T14:01:13.5164142-03:00 PureWebSocket.Send: Adding item to send queue: Data {"event":"#handshake","data":{"authToken":null},"cid":1}
2018-05-28T14:01:13.5208287-03:00 PureSocketClusterSocket.Emit: Emit invoked, Event auth, Object PureSocketCluster.Creds.
2018-05-28T14:01:13.5212973-03:00 PureWebSocket.StartSender: Sending message: 5/28/2018 5:01:13 PM {"event":"#handshake","data":{"authToken":null},"cid":1}.
2018-05-28T14:01:13.6776999-03:00 PureWebSocket.Send: Adding item to send queue: Data {"event":"auth","data":{"apiKey":"21ae3095b9c74xxxx","apiSecret":"159ad871351b46bd9xxxx"}}
2018-05-28T14:01:13.6778264-03:00 PureWebSocket.StartSender: Sending message: 5/28/2018 5:01:13 PM {"event":"auth","data":{"apiKey":"21ae3095b9c74de0xxxxxxx","apiSecret":"159ad871351b46bdxxxxxx"}}.
2018-05-28T14:01:13.6784274-03:00 PureWebSocket.StartListener: Message fully received: {"rid":1,"data":{"id":"d_dtgBfTkLJsuamvACnL","pingTimeout":20000,"isAuthenticated":false}}
2018-05-28T14:01:13.6984465-03:00 PureSocketClusterSocket.Socket_OnMessage: Received message: {"rid":1,"data":{"id":"d_dtgBfTkLJsuamvACnL","pingTimeout":20000,"isAuthenticated":false}}
2018-05-28T14:01:13.8647483-03:00 PureSocketClusterSocket.SubscribeChannels: SubscribeChannels invoked.
2018-05-28T14:01:13.8702702-03:00 PureWebSocket.StartListener: Message fully received: {"event":"#disconnect","data":{"code":1000}}
2018-05-28T14:01:13.8705941-03:00 PureSocketClusterSocket.Socket_OnMessage: Received message: {"event":"#disconnect","data":{"code":1000}}
2018-05-28T14:01:13.8790469-03:00 PureWebSocket.StartListener: Server requested close.
2018-05-28T14:01:13.8833737-03:00 PureWebSocket.StartListener: Binary fully received:
2018-05-28T14:01:13.8848716-03:00 PureSocketClusterSocket.Socket_OnData: Received data:
2018-05-28T14:01:13.8849724-03:00 PureWebSocket.StartListener: Listener exiting
```

**Comment by ByronAP** [comment]
Are you using the latest release/demo?

**Comment by Vannazux** [comment]
Yes, I am. 

Sorry, It just worked after last try now. I don`t know why it was getting stuck authenticating before. 

The problem I have now after uncommenting part of the code `var channels = _client.GetChannels();`  from the method  `Client_OnClientReady()` is that it throws a NullReference exception now and seems not to return the available channel list anymore. Should I do anything different to get this list?

**Comment by Vannazux** [comment]
It seems that the methods `GetChannels()` and `GetExchanges()` in CoinigyWSClient class always return the first case scenario (null) for the `if (_scClient == null || _scClient.SocketState != WebSocketState.Open || !_ready) ` 



`Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'Cannot convert type 'System.Collections.Generic.KeyValuePair<string,object>' to 'System.Collections.Generic.Dictionary<string,object>''`

**Comment by ByronAP** [comment]
Good catch, thank you, I will issue a fix shortly

**Comment by ByronAP** [comment]
fixed in v2.0.2
